### DISCUSSION TITLE
Standard Open API Rendering

### DISCUSSION THREAD

**Discussion by rwithers** [go-fuego/fuego#154] [Q&A]
Hello, 

  I may be missing something but I can't seem to figure out how to get the open api spec to render without the stoplight wrapper.  Seems like hello-world@latest has it but I don't know that I've found a reference to that.  

I will keep looking and respond if I find it.  

Any guidance would be appreciated. 

Thanks, 

Ryan

**Comment by dylanhitt** [comment]
Take a look at the [WithOpenAPIConfig](https://pkg.go.dev/github.com/go-fuego/fuego#WithOpenAPIConfig) option. There's a field UIHandler where you can pass your handler.

**Reply by dylanhitt** [reply]
It appears if you set the field to `nil` the server will produce an empty response. 🤔 Probably don't even want the route register in that case. Also causes a panic.

**Reply by rwithers** [reply]
Yeah I'm curious if you run the 

`go run github.com/go-fuego/fuego/examples/hello-world@latest`

It comes back with swagger running without the stoplight rendering around it all.  That's what I'm after.  

The other alternative would be some guidance on overriding the default landing page for the stoplight stuff.  

I definitely need to keep reading.

**Reply by dylanhitt** [reply]
Oh I see now. 

That's because `go run github.com/go-fuego/fuego/examples/hello-world@latest` is building an older version. If you 

```
go run github.com/go-fuego/fuego/examples/hello-world@main
```

It will be the stoplight UI. I imagine a reason for the switch was to remove the `httpswagger` dep from fuego. I was able to get the the base swagger ui working with something like this

```golang
package main

import (
	"net/http"

	"github.com/go-fuego/fuego"
	httpSwagger "github.com/swaggo/http-swagger"
)

func openApiHandler(specURL string) http.Handler {
	return httpSwagger.Handler(
		httpSwagger.Layout(httpSwagger.BaseLayout),
		httpSwagger.PersistAuthorization(true),
		httpSwagger.URL(specURL), // The url pointing to API definition
	)
}

func main() {
	s := fuego.NewServer(
		fuego.WithOpenAPIConfig(fuego.OpenAPIConfig{
			UIHandler: openApiHandler,
		}),
	)
        s.OpenApiSpec.OpenAPI = "3.0.3"

	fuego.Get(s, "/", helloWorld)

	s.Run()
}

func helloWorld(c fuego.ContextNoBody) (string, error) {
	return "Hello, World!", nil
}

```

One thing to note is that `swaggo/http-swagger` package doesn't support openapi 3.1 hence why you have to override the openapi version at least starting with fuego `> 0.14`

Hopefully that helps you out

**Reply by EwenQuim** [reply]
@rwithers You may have missed it but it was in [the docs](https://go-fuego.github.io/fuego/docs/guides/openapi#custom-ui)! 

<img width="1364" alt="image" src="https://github.com/user-attachments/assets/1b0e8460-9106-49e8-99a6-51b51e2a2b94">
 I'm curious why you don't like the Stoplight interface? Can't wait to find out, maybe we chose the wrong default option?

---
@dylanhitt I still see different actions we could do here

- Having a script to automatically bump examples with new tags, so each time a new Fuego version is out, the examples are updated at @latest and not only @main
- Making it clear in the docs that we can just write this to disable OpenAPI config

```go
app := fuego.NewServer(
	fuego.WithOpenAPIConfig(fuego.OpenAPIConfig{
		DisableSwagger:   os.Getenv("ENV") != "dev",
		DisableLocalSave: os.Getenv("ENV") != "dev",
	}),
)
```

- Mark the version as 3.0.x instead of 3.1, I didn't think that `swaggo/http-swagger` and other tools might no support openapi3.0-friendly openapi3.1 specs.
- Create another example without using `swaggo/http-swagger` but using a std custom handler to make it more clear how one could do this

What do you think about it ?

**Reply by dylanhitt** [reply]
> Making it clear in the docs that we can just write this to disable OpenAPI config

Yeah, I did just realize the DisableSwagger doesn't necessarily stop the saving of the doc. Which it probably should.. I'll make PR if you agree.

> Create another example without using swaggo/http-swagger but using a std custom handler to make it more clear how one could do this

I think is the best course of action

**Reply by dylanhitt** [reply]
@rwithers was this answered? Or can we provide some more clarification.
